server/upgrade.js

```diff
@@ -0,0 +1,71 @@
+const mongo   = require("promised-mongo")
+const db      = mongo("mongodb://localhost/bydesign", ["authors", "posts", "metadata", "categories"])
+const logger  = require("beautiful-log")
+const config  = require("../config")
+const utils   = require("./utils")
+
+let isLessThan = (desired, upgrader) => 
+  version => version <= desired ? upgrader() : version
```

Comment:
I think parens around the argument `version` in the arrow function would be more readable, i.e.

```
(version) => version <= desired ? upgrader() : version
```

It took me longer than I'd care to admit to realize that the first `version` was an argument.

Alternatively, move the curried argument up with the first set of arguments (or even do both).